tests/lib/IO/Flysystem/PathPrefixer/BaseSiteAccessAwarePathPrefixerTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Tests\Core\IO\Flysystem\PathPrefixer;
+
+use Ibexa\Core\IO\Flysystem\PathPrefixer\PathPrefixerInterface;
+use Ibexa\Tests\Core\Search\TestCase;
+
+/**
+ * @covers \Ibexa\Core\IO\Flysystem\PathPrefixer\PathPrefixerInterface
+ */
+abstract class BaseSiteAccessAwarePathPrefixerTest extends TestCase
+{
+    abstract protected function getPrefixer(): PathPrefixerInterface;
+
+    abstract public function getDataForTestPrefixPath(): iterable;
+
+    abstract public function getDataForTestPrefixDirectoryPath(): iterable;
+
+    abstract public function getDataForTestStripPrefixPath(): iterable;
+
+    public function getDataForTestStripDirectoryPrefix(): iterable
```

Comment:
What data  shapes are expected?

```suggestion
    /**
     * @return iterable<string, array{string, string}>
     */
    final public function getDataForTestStripDirectoryPrefix(): iterable
```

This includes the data providers as well.